mailbox/store/src/main/java/org/apache/james/mailbox/store/search/SearchUtil.java

```diff
@@ -171,6 +173,18 @@ public static String getMailboxAddress(String headerValue) {
         }
         return "";
     }
+
+
+    public static String getSerializedMessageIdIfSupportedByUnderlyingStorageOrNull(MailboxMessage mailboxMessage) {
+        try {
+            MessageId messageId = mailboxMessage.getMessageId();
+            if (messageId != null) {
+                return messageId.serialize();
+            }
```

Comment:
I think there is still an issue here. When you see the Message interface, you have no idea that you have to do a null check on this before using getMessageId() or that you need to use this getSerializedMessageIdIfSupportedByUnderlyingStorageOrNull method.
If you're assuming that getMessageId() can return null, please don't and make it Optional<MessageId>